app/src/main/java/com/example/whatsapp_clone/Views/MainActivity.java

```diff
@@ -141,10 +147,129 @@ public boolean onQueryTextChange(String newText) {
+        //USER SECTION
+        View userDetails = bottomSheetView.findViewById(R.id.current_user_details);
+        CircleImageView userImage = bottomSheetView.findViewById(R.id.currentUserImg);
+        TextView displayName = bottomSheetView.findViewById(R.id.tvUserDisplayName);
+        User currentUser = Repository.getInstance().getCurrentUser();
+        if(currentUser != null) {
+            userDetails.setVisibility(View.VISIBLE);
+            userImage.setImageBitmap(Utils.getDecodedPic(currentUser.profilePic));
+            displayName.setText(currentUser.displayName);
+        }else {
+            userDetails.setVisibility(View.GONE);
+        }
+
+        //Dark mode section
+        SwitchCompat darkModeSwitch = (SwitchCompat) bottomSheetView.findViewById(R.id.switchDarkMode);
+
+        darkModeSwitch.setOnCheckedChangeListener((buttonView, isChecked) -> {
+            if(isChecked) {
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
+            } else {
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
+            }
+        });
+
+        // setURL
+
+        View setUrlView = bottomSheetView.findViewById(R.id.llChangeApi);
+        EditText urlET = bottomSheetView.findViewById(R.id.set_url_et);
+        setUrlView.setOnClickListener(view -> {
+            if(urlET.getVisibility() == View.GONE) {
+                urlET.setVisibility(View.VISIBLE);
+                urlET.setAlpha(0f);
+                urlET.animate()
+                        .alpha(1f)
+                        .setDuration(150)
+                        .setListener(null);
+            } else if(urlET.getVisibility() == View.VISIBLE) {
+                urlET.animate()
+                        .alpha(0f)
+                        .setDuration(150)
+                        .setListener(new AnimatorListenerAdapter() {
+                            @Override
+                            public void onAnimationEnd(Animator animation) {
+                                //todo: did not handled url validation
+                                Repository.getInstance()
+                                                .setBaseURL(urlET.getText().toString());
+                                urlET.setVisibility(View.GONE);
+                            }
+                        });
+            }
+        });
+
+        //Logout view
+        View logoutView = bottomSheetView.findViewById(R.id.logoutLayout);
+        logoutView.setOnClickListener(view -> {
+            Repository.getInstance().logOut();
+            dialog.dismiss();
+
+            NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
+            int firstDestinationId = navController.getGraph().getStartDestination();
+            navController.popBackStack(firstDestinationId, false);
+        });
+
+
+
+    }
```

Comment:
there is a lot of logic in the main activity, should we move some of it to a utility class or something?

Reply:
this is exactly the job for the main activity. to handle global changes for the entire app, such as pressing a button on the menu and do some UI staff, no extra logic is here